### PR TITLE
ensure deserialization is agnostic to serialization of field order

### PR DESCRIPTION
**Pull request by jrevels**
should be mergeable once https://github.com/apache/arrow-julia/pull/493 is released

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,18 +1,20 @@
 name = "Legolas"
 uuid = "741b9549-f6ed-4911-9fbf-4a1c0c97f0cd"
 authors = ["Beacon Biosignals, Inc."]
-version = "0.5.16"
+version = "0.6.0"
```
Comment (ararslan) [inline Project.toml:4]: Seems like this would constitute a bugfix rather than a breaking release?

Comment (jrevels) [reply Project.toml:4]: it introduces new higher compat bounds

Comment (ericphanson) [reply Project.toml:4]: That’s not considered breaking since Pkg will handle it and not install this version if you aren’t compatible

Comment (jrevels) [reply Project.toml:4]: okey dokey

Comment (jrevels) [reply Project.toml:4]:
```suggestion
version = "0.5.17"
```

**Pull request merged**